lms/djangoapps/program_enrollments/models.py

```diff
@@ -196,3 +188,30 @@ def change_status(self, status):
             ))
         self.save()
         return self.status
+
+    def enroll(self, user):
+        """
+        Create a StudentCourseEnrollment to enroll user in course
+        """
+        try:
+            self.course_enrollment = StudentCourseEnrollment.enroll(
+                user,
+                self.course_key,
+                mode=CourseMode.MASTERS,
+                check_access=True,
+            )
+        except AlreadyEnrolledError:
+            course_enrollment = StudentCourseEnrollment.objects.get(
+                user=user,
+                course_id=self.course_key,
+            )
+            if course_enrollment.mode == CourseMode.AUDIT or course_enrollment.mode == CourseMode.HONOR:
+                course_enrollment.mode = CourseMode.MASTERS
+                course_enrollment.save()
+            self.course_enrollment = course_enrollment
+            message = ("Attempted to create course enrollment for user={user} and course={course}"
+                       " but an enrollment already exists. Existing enrollment will be used instead")
+            logger.info(message.format(user=user.id, course=self.course_key))
+        if self.status == CourseEnrollmentResponseStatuses.INACTIVE:
```

Comment:
We should consider renaming (or aliasing the import) `CourseEnrollmentResponseStatuses` to `ProgramCourseEnrollmentResponseStatus` to help disambiguate it from a "regular" `CourseEnrollment`.